New translation: gatsby-config.md (#182)
gatsbyjs/gatsby-pt-BR: docs/docs/gatsby-config.md

---
title: API de configuração do Gatsby
---

As opções de configurações para um site Gatsby são colocadas em um arquivo na raiz do projeto chamado `gatsby-config.js`.

> _Anotações: Existem muitas amostras de configurações que podem ser úteis para referenciar em diferentes [Exemplos de Websites Gatsby](https://github.com/gatsbyjs/gatsby/tree/master/examples).

Reviewer: ```suggestion
_Anotações: Existem muitas amostras de configurações que podem ser úteis para referenciar em diferentes [Exemplos de Websites Gatsby](https://github.com/gatsbyjs/gatsby/tree/master/examples)._
```

Reviewer: Apenas para fechar o itálico aberto no inicio.

## siteMetadata

Quando você quer reutilizar alguns pedaços comuns através do site (por exemplo, o título do site), você pode armazenar estes dados em `siteMetadata`:

```javascript:title=gatsby-config.js
module.exports = {
  siteMetadata: {
    title: `Gatsby`,
    siteUrl: `https://www.gatsbyjs.org`,
    description: `Gerador de site moderno e veloz para React`,
  },
}
```

> Desta forma, você pode armazenar em um único local e puxar isto sempre que precisar, se você sempre precisar atualizar as informações, você somente tem que mudar aqui.

Reviewer 2: ```suggestion
Desta forma, você pode armazenar em um único local e puxar isto sempre que precisar. Se você sempre precisar atualizar as informações, você somente tem que mudar aqui.
```

Veja a descrição completa e exemplo de uso em [Tutorial Gatsby.js Parte Quatro](/tutorial/part-four/#data-in-gatsby).

## Plugins

> Plugins são pacotes Node.js que implementam APIs do Gatsby. Os arquivos de configurações aceitos em um array de plugins. Alguns plugins podem ser listados somente pelo nome, enquanto outros podem precisar de opções (veja a documentação para plugins individuais).

Reviewer 2: ```suggestion
Plugins são pacotes Node.js que implementam APIs do Gatsby. O arquivo de configurações aceita um array de plugins. Alguns plugins podem ser listados somente pelo nome, enquanto outros podem precisar de opções (veja a documentação para plugins individuais).
```

}
```

> Veja mais sobre [Plugins](/docs/plugins/) para saber mais como utilizar plugins, e ver plugins oficiais disponíveis e plugins da comunidade.

Reviewer 2: ```suggestion
Veja mais sobre [Plugins](/docs/plugins/) para saber mais como utilizá-los, além de conferir os plugins oficiais e os feitos pela comunidade.
```

## pathPrefix

> É comum sites serem hospedados em algum outro local que não seja domínio raiz. Quando dizemos que temos um Gatsby site em `example.com/blog/`. Nesse caso, precisamos de um prefixo (`/blog`) adicionado para todos os caminhos no site.

Reviewer 2: ```suggestion
É comum sites serem hospedados em algum outro local que não seja domínio raiz. Quando dizemos que temos um site Gatsby em `example.com/blog/`. Nesse caso, precisamos de um prefixo (`/blog`) adicionado para todos os caminhos no site.
```

```javascript:title=gatsby-config.js
module.exports = {
  pathPrefix: `/blog`,
}
```

Veja mais sobre [Adicionar um Prefixo de Caminho](/docs/path-prefix/).

## Polyfill

> Gatsby usa a API do ES6 Promise. Pelo fato de alguns navegadores não suportarem isso, o Gatsby inclui um polyfill Promise por padrão.

Reviewer 2: ```suggestion
Gatsby usa a API do ES6 Promise. Pelo fato de alguns navegadores não suportarem isso, o Gatsby inclui uma _Promise_ polyfill por padrão.
```

> Se você desejar fornecer seu próprio polyfill Promise, você pode configurar o `polyfill` como false.

Reviewer 2: ```suggestion
Se você desejar fornecer sua própria _Promise_ polyfill, você pode configurar o `polyfill` como _false_.
```

```javascript:title=gatsby-config.js
module.exports = {
  polyfill: false,
}
```

Veja mais sobre [Suporte de Navegadores](/docs/browser-support/#polyfills) em Gatsby.

## Mapeando tipos de nó

O Gatsby inclui um recurso avançado que permite criar "mappings" entre os tipos de nós.

> Anotações: Gatsby v2.2 introduziu uma nova maneira de criar relações de chave estrangeira entre os tipos de nós com [o `@link` Extensão de campo GraphQL](/docs/schema-customization/#foreign-key-fields).

> Por exemplo, imagine que você tem um blog de remarcação com vários autores no qual deseja "vincular" de cada postagem do blog às informações do autor armazenadas em um arquivo yaml chamado `author.yaml`:

Reviewer 2: ```suggestion
Por exemplo, imagine que você tem um blog em markdown com vários autores no qual deseja "vincular" cada postagem do blog às informações do autor armazenadas em um arquivo yaml chamado `author.yaml`:
```

}
```

Pode ser necessário instalar o transformador de arquivo apropriado (neste caso [YAML](/packages/gatsby-transformer-yaml/)) e configurar o [gatsby-source-filesystem](/packages/gatsby-source-filesystem/) adequadamente para que o Gatsby pegue os arquivos de mapeamento. Isso se aplica a outros tipos de arquivos mencionados posteriormente neste segmento também.

> Gatsby usa esse mapeamento ao criar o GraphQL esquema para permitir a consulta de dados de ambas as fontes. 

Reviewer 2: ```suggestion
Gatsby usa esse mapeamento ao criar um schema GraphQL para permitir a consulta de dados de ambas as fontes. 
```

]
```

> E adicione a seguinte regra ao seu `gatsby-config.js`:

Reviewer 2: ```suggestion
E você adiciona a seguinte regra ao seu `gatsby-config.js`:
```

}
```

De maneira semelhante aos arquivos YAML e JSON, o mapeamento entre arquivos Markdown também pode ser usado para mapear um array de ids.

## Proxy

> Definir a opção de configuração do proxy informará o servidor de revelação como proxy de quaisquer solicitações desconhecidas para o servidor especificado. Por exemplo:

Reviewer 2: ```suggestion
Definir a opção de configuração do proxy informará o servidor de desenvolvimento para realizar proxy de qualquer solicitação desconhecida endereçada a um servidor especificado. Por exemplo:
```

```javascript:title=gatsby-config.js
module.exports = {
  proxy: {
    prefix: "/api",
    url: "http://examplesite.com/api/"
  }

> };

Reviewer 2: ```suggestion
}
```
